src/ui/common/src/components/tables/KeyValueTable.tsx

```diff
@@ -0,0 +1,101 @@
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell, { TableCellProps } from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import * as React from 'react';
+
+import { Data, DataSchema } from '../../utils/data';
+import { CheckTableItem } from './CheckTableItem';
+
+export enum KeyValueTableType {
+  Metric = 'metric',
+  Check = 'check',
+}
+
+interface KeyValueTableProps {
+  rows: Data;
+  schema?: DataSchema;
+  height?: string;
+  width?: string;
+  maxHeight?: string;
+  stickyHeader?: boolean;
+  tableAlign?: string;
+  tableType: KeyValueTableType;
+}
+
+const kvSchema: DataSchema = {
+  fields: [
+    { name: 'Title', type: 'varchar' },
+    { name: 'Value', type: 'varchar' },
+  ],
+  pandas_version: '0.0.1', // TODO: Figure out what to set this value to.
+};
+
+export const KeyValueTable: React.FC<KeyValueTableProps> = ({
+  rows,
+  schema = kvSchema,
+  height = '440px',
+  width = '100%',
+  maxHeight = '440px',
+  stickyHeader = true,
+  tableAlign = 'left',
+  tableType,
+}) => {
+  return (
+    <TableContainer sx={{ maxHeight, height, width }}>
+      <Table
+        stickyHeader={stickyHeader}
+        aria-label={stickyHeader ? 'sticky table' : 'table'}
+      >
+        <TableHead>
+          <TableRow>
+            {schema.fields.map((column, idx) => (
+              <TableCell
+                key={`${column.name}-heading-${idx}`}
+                align={tableAlign as TableCellProps['align']}
+              >
+                {column.name}
+              </TableCell>
+            ))}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {rows.data.map((row, rowIndex) => {
+            return (
+              <TableRow
+                hover
+                role="checkbox"
+                tabIndex={-1}
+                key={`tableBody-${rowIndex}`}
+              >
+                {schema.fields.map((column, columnIndex) => {
+                  const value = row[columnIndex];
+
+                  // For title columns we should just render the text.
+                  // For a check's value column, we should render the appropriate icon.
+                  return (
+                    <TableCell
+                      key={`cell-${rowIndex}-${columnIndex}`}
+                      align={tableAlign as TableCellProps['align']}
+                    >
+                      {tableType === KeyValueTableType.Metric ||
+                      column.name === 'Title' ? (
+                        value
+                      ) : (
+                        <CheckTableItem checkValue={value as string} />
```

Comment:
Maybe we should rename this component from KeyValueTable to something less generic.

At the moment what this table does is show metric and check names next to their execution status. perhaps OperatorExecStateTable may be a better name?

Reply:
✅